packages/components/src/mobile/bottom-sheet/index.native.js

```diff
@@ -435,7 +431,7 @@ class BottomSheet extends Component {
 
 		let listStyle = {};
 		if ( isFullScreen ) {
-			listStyle = { flexGrow: 1 };
```

Comment:
This is the core issue:
The Flatlist is not scrollable with the `FlexGrow` property. 
Instead we need to set the `maxHeight` property, which requires the changes made in `onSetMaxHeight()` mentioned above